web-client/public/js/setup-load-and-import-handlers.js

```diff
@@ -0,0 +1,152 @@
+
+const submittedFilename = $upload => {
+    let path = $upload.val();
+    let fakePathCheck = path.search("\\\\") + 1;
+
+    while (fakePathCheck) {
+        path = path.substring(fakePathCheck);
+        fakePathCheck = path.search("\\\\") + 1;
+    }
+
+    return path;
+};
+
+const createFileForm = $upload => {
+    const formData = new FormData();
+    formData.append("file", $upload[0].files[0]);
+    return formData;
+};
+
+const uploadEpilogue = event => {
+    if (window.ga) {
+        window.ga("send", "pageview", {
+            page: "/GRNsight/upload",
+            sessionControl: "start"
+        });
+    }
+
+    $("a.upload > input[type=file]").val("");
+    event.preventDefault();
+};
+
+const uploadHandler = (uploadRoute, uploader) => {
+    return function (event) { // Must be `function` due to use of `this`.
+        const $upload = $(this);
+        const filename = submittedFilename($upload);
+        const formData = createFileForm($upload);
+        uploader(uploadRoute, filename, formData);
+        uploadEpilogue(event);
+    };
+};
+
+const networkErrorDisplayer = xhr => {
+    // Deleted status, error for argument because it was never used
+    const err = JSON.parse(xhr.responseText);
+    let errorString = "Your graph failed to load.<br><br>";
+
+    if (!err.errors) { // will be falsy if an error was thrown before the network was generated
+        errorString += err;
+    } else {
+        console.log(err.errors);
+        errorString = err.errors.reduce(
+            (currentErrorString, currentError) =>
+                `${currentErrorString}${currentError.possibleCause} ${currentError.suggestedFix}<br><br>`,
+
+            errorString
+        );
+    }
+
+    $("#error").html(errorString);
+    $("#errorModal").modal("show");
+};
+
+let reloader = () => { };
+
+export const setupLoadAndImportHandlers = grnState => {
+    const loadGrn = (url, name, formData) => {
+        // The presence of formData is taken to indicate a POST.
+        const fullUrl = [ $("#service-root").val(), url ].join("/");
+        (formData ? $.ajax({
+            url: fullUrl,
+            data: formData,
+            processData: false,
+            contentType: false,
+            type: "POST",
+            crossDomain: true
+        }) : $.getJSON(fullUrl)).done((network, textStatus, jqXhr) => {
+            // Display the network in the console
+            console.log(network);
+            grnState.name = name || jqXhr.getResponseHeader("X-GRNsight-Filename");
+            grnState.network = network;
+
+            reloader = () => loadGrn(url, name, formData);
+
+            updateApp(grnState);
+            // displayStatistics(network);
+        }).error(networkErrorDisplayer);
+    };
+
+    /*
+     * Thanks to http://stackoverflow.com/questions/6974684/how-to-send-formdata-objects-with-ajax-requests-in-jquery
+     * for helping to resolve this.
+     */
+
+    // TODO Some opportunity for unification with loadGrn?
```

Comment:
Another worthy `TODO`. There is definitely some similarity between `loadGrn` and `importGrn`. Consolidating those into one function will give us a nice little reduction in code.